bundles/core/org.openhab.core/src/main/java/org/openhab/core/types/UnDefType.java

```diff
@@ -34,4 +34,20 @@ public String format(String pattern) {
 		return String.format(pattern, this.toString());
 	}
 
+	/**
+	 * Returns the enum constant with the specified name.
+	 */
+	public static UnDefType parse(String name) {
+		if (name == null) {
+			throw new NullPointerException("Name is null");
+		}
+		if (UNDEF.toString().equals(name)) {
+			return UNDEF;
+		} else if (NULL.toString().equals(name)) {
+			return NULL;
+		}
+		throw new IllegalArgumentException("No enum constant "
+				+ UnDefType.class.getCanonicalName() + "." + name);
+	}

```

Comment:
I can't override the valueOf method in an enum:
```
The enum UnDefType already defines the method valueOf(String) implicitly
```

Try it:
```
UnDefType.valueOf("NULL"):	Uninitialized
UnDefType.valueOf("UNDEF"):	Undefined

UnDefType.valueOf(UnDefType.NULL.toString()): IllegalArgumentException: No enum constant org.openhab.core.types.UnDefType.Uninitialized
UnDefType.valueOf(UnDefType.UNDEF.toString()): IllegalArgumentException: No enum constant org.openhab.core.types.UnDefType.Undefined
```
The TypeParser never worked with the UnDefType

I found this problem while i wrote a jsr223 script, i wanted to update a item:
```
postUpdate("ITEM_NAME", UnDefType.UNDEF);
```
i got the IllegalArgumentException and searched the reason